develop: review bookshop schema design (#8642)
pingcap/docs: develop/dev-guide-bookshop-schema-design.md

# Bookshop Example Application

Bookshop is a virtual online bookstore application through which you can buy books of various categories and rate the books you have read.

> To make your reading on the application developer guide more smoothly, we present the SQL statements based on the [table structures](#description-of-the-tables) and data of the Bookshop application. This document focuses on the methods of importing the table structures and data as well as the definitions of the table structures.

Reviewer: ```suggestion
To make your reading on the application developer guide more smoothly, we present the example SQL statements based on the [table structures](#description-of-the-tables) and data of the Bookshop application. This document focuses on the methods of importing the table structures and data as well as the definitions of the table structures.
```

## Import table structures and data

You can import Bookshop table structures and data either [via TiUP](#via-tiup-demo) or [via the import feature of TiDB Cloud](#via-tidb-cloud-import).

<SimpleTab>
<div label="Via `TiUP demo`">

### Via `tiup demo`

If your TiDB cluster is deployed using [TiUP](/tiup/tiup-reference.md#tiup-reference) or you can connect to your TiDB server, you can quickly generate and import sample data for the Bookshop application by running the following command:

{{< copyable "shell" >}}

```shell
tiup demo bookshop prepare
```

> By default, this command enables your application to connect to port `4000` on address `127.0.0.1`, enables you to log in as the root user without a password, and creates a [table structure](#description-of-the-tables) in the database named `bookshop`.

Author: ```suggestion
By default, this command enables your application to connect to port `4000` on address `127.0.0.1`, enables you to log in as the `root` user without a password, and creates a [table structure](#description-of-the-tables) in the database named `bookshop`.
```

# Bookshop Example Application

Bookshop is a virtual online bookstore application through which you can buy books of various categories and rate the books you have read.

> This document describes the following information:

Reviewer: You can refer to new changes on the chinese document, our website already has a page navigation feature, no need to add redundant outlines.

https://github.com/pingcap/docs-cn/blob/e02b2eab472ac67ab2a9d470dcf2e00447204cc8/develop/dev-guide-bookshop-schema-design.md#bookshop-%E5%BA%94%E7%94%A8

- How to import data into the Bookshop application via TiUP and the import function of TiDB Cloud
- What are the definitions of the tables in the Bookshop application
- How to use the `dbinit.sql` script

## Import data

You can import data either by using `tiup demo` or by using the import function of TiDB Cloud.

<SimpleTab>
<div label="Via `TiUP demo`">

### Via `tiup demo`

> If your TiDB cluster is deployed using [TiUP](/tiup/tiup-reference.md#tiup-reference) or you can connect to your TiDB server, you can quickly generate and import sample data for the Bookshop application by running the following command:

Reviewer 2: ```suggestion
If your TiDB cluster is deployed using [TiUP](/tiup/tiup-reference.md#tiup-reference) and you can connect to your TiDB server, you can quickly generate and import sample data for the Bookshop application by running the following command:
```

Author: @Mini256 Could you help confirm this modification. In the Chinese version, the word is "或者”, indicating either situation applies.

Reviewer: use `or`, this means that if you use TIDB Cloud to start the TIDB cluster, you can use this command as long as you can connect.

{{< copyable "shell" >}}

```shell
tiup demo bookshop prepare
```

> By default, this command connects to port `4000` on address `127.0.0.1`, enables you to log in as the root user without a password, and creates a [table structure](#description-of-the-tables) in a database named `bookshop`.

Reviewer 2: ```suggestion
By default, this command enables your application to connect to port `4000` on address `127.0.0.1`, enables you to log in as the root user without a password, and creates a [table structure](#description-of-the-tables) in a database named `bookshop`.
```

| `--password` | `-p` | None        | Database user password |
| `--db`       | `-D` | `bookshop`  | Database name     |

For example, if you want to connect to a database on TiDB Cloud, you can specify the connection information as follows:

{{< copyable "shell" >}}

```shell
tiup demo bookshop prepare -U root -H tidb.xxx.yyy.ap-northeast-1.prod.aws.tidbcloud.com -P 4000 -p
```

#### Set the data volume

You can specify the volume of data to be generated in each database table by configuring the following parameters:

| Parameter        | Default value   | Description                              |
| ----------- | -------- | --------------------------------- |
| `--users`   | `10000`  | The number of rows of data to be generated in the `users` table   |
| `--authors` | `20000`  | The number of rows to be generated in the `authors` table |
| `--books`   | `20000`  | The number of rows of data to be generated in the `books` table   |
| `--orders`  | `300000` | The number of rows of data to be generated in the `orders` table  |
| `--ratings` | `300000` | The number of rows of data to be generated in the `ratings` table |

For example, the following command is executed to generate:

> - 200,000 rows of user information via the `--users` parameter

Reviewer 2: 位数每隔三位加逗号，主要用在货币数字上：https://developers.google.com/style/numbers

   **Role-ARN**:

    {{< copyable "" >}}

    ```
    arn:aws:iam::494090988690:role/s3-tidb-cloud-developer-access
    ```

    In this example, the following data is generated in advance:

    - 200,000 rows of user information
    - 500,000 rows of book information
    - 100,000 rows of author information
    - 1,000,000 rows of rating records
    - 1,000,000 rows of order records

2. Select **US West (Oregon)** for **Bucket Region**.
3. Select **TiDB Dumpling** for **Data Format**.

    ![Import Bookshop data in TiDB Cloud](/media/develop/tidb_cloud_import_bookshop_data.png)

4. Enter database login information.

> 5. Click the **Import** option to confirm the import.

Reviewer: I think it's a button, not an option.

Author: > I think it's a button, not an option.
Please see the word list of [Google developer documentation style guide](https://developers.google.cn/style/word-list#letter-b)
![image](https://user-images.githubusercontent.com/89768198/169734375-711f3f75-ae9e-40da-9f1a-00f6aad01c4c.png)

Reviewer: Let us look at this guide in conjunction with practical examples:

https://cloud.google.com/s/results?q=option
https://cloud.google.com/s/results?q=button

I think this document ignores the button description of the PC side, and I believe **Import** is an on-screen button.

6 rows in set (0.03 sec)
```

</div>
</SimpleTab>

> ## Description of the Tables

Reviewer 2: ```suggestion
## Description of the tables
```

| Field name | Type       | Description                                                      |
|------------|------------|--------------------------------------------------------------|
| book_id    | bigint(20) | Unique ID of a book (linked to [books](#books-table))      |
| author_id  | bigint(20) | Unique ID of an author（Link to [authors](#authors-table)） |

### `orders` table

This table stores user purchase information.

| Field name | Type       | Description                                                        |
|------------|------------|----------------------------------------------------------------|
| id         | bigint(20) | Unique ID of an order                                     |
| book_id    | bigint(20) | Unique ID of a book (linked to [books](#books-table))        |
| user_id    | bigint(20) | User unique identifier (associated with [users](#users-table)) |
| quantity   | tinyint(4) | Purchase quantity                                              |
| ordered_at | datetime   | Purchase time                                                  |

> ## Database initialization script `dbinit.sql`

Reviewer: The new section description needs to be translated:

https://github.com/pingcap/docs-cn/blob/e02b2eab472ac67ab2a9d470dcf2e00447204cc8/develop/dev-guide-bookshop-schema-design.md#%E6%95%B0%E6%8D%AE%E5%BA%93%E5%88%9D%E5%A7%8B%E5%8C%96-dbinitsql-%E8%84%9A%E6%9C%AC
